src/jace/stages.py

```diff
@@ -256,64 +276,127 @@ def as_sdfg(self) -> dace.SDFG:
-        csdfg: The compiled SDFG object.
-        inp_names: Names of the SDFG variables used as inputs.
-        out_names: Names of the SDFG variables used as outputs.
+        compiled_sdfg: The compiled SDFG object.
+        input_names: SDFG variables used as inputs.
+        output_names: SDFG variables used as outputs.
+        out_tree: Pytree describing how to unflatten the output.
 
     Note:
         The class assumes ownership of its input arguments.
 
     Todo:
-        - Handle pytrees.
+        - Automatic strides adaptation.
     """
 
-    _csdfg: dace_helper.CompiledSDFG
-    _inp_names: tuple[str, ...]
-    _out_names: tuple[str, ...]
+    _compiled_sdfg: tjsdfg.CompiledJaxprSDFG
+    _out_tree: jax_tree.PyTreeDef
 
     def __init__(
-        self, csdfg: dace_helper.CompiledSDFG, inp_names: Sequence[str], out_names: Sequence[str]
+        self,
+        compiled_sdfg: tjsdfg.CompiledJaxprSDFG,
+        out_tree: jax_tree.PyTreeDef,
     ) -> None:
-        if (not inp_names) or (not out_names):
-            raise ValueError("Input and output can not be empty.")
-        self._csdfg = csdfg
-        self._inp_names = tuple(inp_names)
-        self._out_names = tuple(out_names)
+        self._compiled_sdfg = compiled_sdfg
+        self._out_tree = out_tree
 
-    def __call__(self, *args: Any, **kwargs: Any) -> Any:
+    def __call__(self, *args: Any, **kwargs: Any) -> _R:
         """
         Calls the embedded computation.
 
-        The arguments must be the same as for the wrapped function, but with
-        all static arguments removed.
+        Note:
+            Unlike the `lower()` function which takes the same arguments as the original
+            computation, to call this function you have to remove all static arguments.
+            Furthermore, all arguments must have strides and storage locations that is
+            compatible with the ones that were used for lowering.
         """
-        return dace_helper.run_jax_sdfg(self._csdfg, self._inp_names, self._out_names, args, kwargs)
+        flat_call_args = jax_tree.tree_leaves((args, kwargs))
+        flat_output = self._compiled_sdfg(flat_call_args)
+        return jax_tree.tree_unflatten(self._out_tree, flat_output)
 
 
-#: Known compilation stages in JaCe.
-Stage = JaCeWrapped | JaCeLowered | JaCeCompiled
+# <--------------------------- Compilation/Optimization options management
+
+_JACELOWERED_ACTIVE_COMPILE_OPTIONS: CompilerOptions = optimization.DEFAULT_OPTIMIZATIONS.copy()
+"""Global set of currently active compilation/optimization options.
+
+The global set is initialized to `jace.optimization.DEFAULT_OPTIMIZATIONS`.
+For modifying the set of active options the the `temporary_compiler_options()`
+context manager is provided.
+To obtain the currently active compiler options use `get_active_compiler_options()`.
+"""
+
+
+@contextlib.contextmanager
+def temporary_compiler_options(new_active_options: CompilerOptions) -> Generator[None, None, None]:
```

Comment:
This is just a comment but I really don't see the need for the `temporary` and `active` adjectives in the name of these functions, since it's obvious from the usage pattern that a context manager is for temporary changes and the only global function getting a set of compiler options is by default returning the currently active options, and otherwise would have another name (e.g. `get_default_compiler_options()`)....

Reply:
I changed them to `set_compiler_options()` and `get_compiler_options()`.